3 - PB/Documentazione interna/Norme di Progetto/Norme di Progetto.typ

```diff
@@ -1697,43 +1697,45 @@ Gli argomenti principali trattati nel documento sono due:
 
 - identificazione dei vincoli implementativi che influenzano i requisiti, l'architettura o il design;
 - realizzazione di un elemento di sistema;
-- l'implementazione viene tracciata.
+- l'implementazione di tale elemento viene tracciata.
 
 === Attività
 
 ==== Preparazione per l'implementazione
-Fondamentale per garantire il livello di qualità richiesto dal #pdq è definire test per l'elemento di sistema che si vuole realizzare secondo il modello di sviluppo _Test Driven Development_. Questo si concretizza nella realizzazione da parte del Progettista di una serie di unit test precedentemente allo sviluppo vero e proprio dell'elemento di sistema. Questo vale anche per codice riutilizzato o codice esterno che viene adattato per soddisfare i requisiti richiesti.
 
-In caso di modifica eseguendo nuovamente i test si garantisce che il software sviluppato e testato in precedenza funzioni ancora come previsto. In caso contrario, si parlerebbe di regressione.
+Fondamentale per garantire il livello di qualità richiesto dal #pdq è definire test specifici per l'elemento di sistema che si vuole realizzare secondo il modello di sviluppo _Test Driven Development_. Questo si concretizza nella realizzazione da parte del Progettista di una serie di unit test precedentemente allo sviluppo vero e proprio dell'elemento di sistema. Questo vale anche per codice riutilizzato o codice esterno che viene adattato per soddisfare i requisiti richiesti.
 
-Data la mole di elementi grafici o interazioni utente che non sono né facili, né economici da testare automaticamente viene definita una modalità di testing manuale: chi sviluppa l'elemento di sistema è responsabile della verifica del corretto funzionamento del codice scritto. Questo vale anche per codice riutilizzato o codice esterno che viene adattato per soddisfare i requisiti richiesti.
+In caso di modifica è necessario reiterare l'esecuzione dei test sviluppati, al fine di garantire che il software sviluppato e testato in precedenza funzioni ancora come previsto. In caso contrario, si parlerebbe di regressione.
 
-Per aiutare il lavoro di verifica da parte del Verificatore, riportare in pr tutte le funzionalità che si ha necessità di controllare manualmente.
+Considerando la complessità degli elementi grafici e delle interazioni utente che non possono essere facilmente o economicamente testati in modo automatico, viene istituita una modalità di testing manuale. In questa modalità, è responsabilità dello sviluppatore dell'elemento di sistema (Programmatore) verificare che il codice scritto funzioni correttamente. Questo vale sia per il codice sviluppato in proprio sia per quello riutilizzato o esterno, che viene adattato per soddisfare i requisiti specifici.
+
+Per aiutare il lavoro di verifica da parte del Verificatore, vengono riportate in pr tutte le funzionalità che si ha necessità di controllare manualmente.
 
 ==== Eseguire l'implementazione
```

Comment:
Questa parte sarebbe da rivedere completamente, perché è troppo generica. Quello che c'è già scritto lo utilizzerei come "cornice", quindi tutto il materiale nuovo lo metti tra quelle due righe che faranno poi da introduzione e conclusione.

Bisogna normare come noi implementiamo il software, quindi alcuni spunti per la rielaborazione potrebbero essere:

- organizzare le categorie di elementi per cartelle, ad esempio tutti i file riguardanti un certo pattern vanno in una cartella dedicata al pattern, tutti quelli della UI in un'altra. Anche ad esempio che dentro ServerActions ci sono effettivamente le Server Actions, ma internamente sono suddivise in cartelle per Zone, Bin, etc. Magari cita anche la sezione 4.5.3.1.3.1 dov'è descritta la repo (dovrebbe avere il tag `@repository-github`) e rimandi alla parte di descrizione di WMS3;
- organizzare il codice per file, quindi un elemento/componente/classe = un file;
- nomi di variabili/metodi/funzioni devono essere parlanti, bisogna capire chi sono e cosa fanno dal nome;
- altre buone pratiche per la codifica.

Reply:
bho onestamente ho un pò improvvisato, sono aperto a suggerimenti su questa